src/librustc_mir/hair/cx/expr.rs

```diff
@@ -656,11 +655,7 @@ fn make_mirror_unadjusted<'a, 'gcx, 'tcx>(cx: &mut Cx<'a, 'gcx, 'tcx>,
 
             // Check to see if this cast is a "coercion cast", where the cast is actually done
             // using a coercion (or is a no-op).
-            let cast = if let Some(&TyCastKind::CoercionCast) =
-                cx.tables()
-                .cast_kinds()
-                .get(source.hir_id)
-            {
+            let cast = if cx.tables().coercion_casts().contains(&source.hir_id.local_id) {
```

Comment:
here the sanity check for `HirId` owners make sense though, so make `coercion_casts` return a  `LocalTableInContext` around the set.

Reply:
```rust
pub struct LocalTableInContext<'a, V: 'a> {
    local_id_root: Option<DefId>,
    data: &'a ItemLocalMap<V>
}
```

How can we make `coercion_casts` return a `LocalTableInContext `? `LocalTableInContext` contains an `ItemLocalMap` whereas  `coercion_casts` is a `ItemLocalSet`

Reply:
hmm... good point. Maybe create an `is_coercion_cast(id: HirId) -> bool` method and call `validate_hir_id_for_typeck_tables` directly inside of it?